src/Mod/Draft/Draft.py

```diff
@@ -706,7 +706,8 @@ def makeCircle(radius, placement=None, face=None, startangle=None, endangle=None
             delta = edge.Curve.Center.sub(placement.Base)
             placement.move(delta)
             if len(edge.Vertexes) > 1:
-                ref = placement.multVec(FreeCAD.Vector(1,0,0))
+                # ref = placement.multVec(FreeCAD.Vector(1,0,0)) ## INCORRECT 
+                ref = FreeCAD.Vector(1,0,0)
```

Comment:
This change would, for example, return an angle of zero if the circle is positioned vertically, which might not be correct... What exactly are you trying to solve here?

Reply:
[](https://forum.freecadweb.org/viewtopic.php?f=10&t=32050&p=298647#p298647)

This was per our discussion on the forums. 

The macro code to generate the issue is:

`import FreeCAD, Part, Draft

# Vectors 
v0 = FreeCAD.Vector(0.0, 0.0, 0.0)
v1 = FreeCAD.Vector(10.0, 0.0, 0.0)
v2 = FreeCAD.Vector(10.0, 10.0, 0.0)

# Wire
Draft.makeWire([v0, v1, v2])

# Arc
arc = Part.ArcOfCircle(v0, v1, v2)
Part.show(arc.toShape())
c = Draft.makeCircle(arc.toShape())
Draft.autogroup(c)`

I understood his complaint as follows:  If he creates the arc directly from Part.ArcOfCircle, the supplied vectors work as expected.  If, however, he tries to generate the arc from Draft.makeCircle and passing the previous Arc shape as the only parameter, it creates the arc with the starting and ending points offset.  His expectation is that passing the arc shape as the only parameter to Draft.makeCircle() should yield an identical arc.  If the reference vector is (0,1,0), it works as expected.

What has me thinking it might be a bug is that, in the conditional, 'isinstance(radius,Part.Edge):', the parameters are derived entirely from the edge itself (apart from the object type).  If that's the case, then it seems as though the result of that segment of code should yield an identical edge.

Reply:
The error was on the calculation of the start and end angle for ArcOfCircle that are not aligned with the Placement.Rotation.

Reply:
Okay thanks for the fix! I think it's right now, so, merging!